lib/manager/cargo/extract.ts

```diff
@@ -63,13 +83,50 @@ function extractFromSection(
 
 export function extractPackageFile(
   content: string,
-  fileName: string
+  fileName: string,
+  config?: ExtractConfig
 ): PackageFile | null {
   logger.trace(`cargo.extractPackageFile(${fileName})`);
-  let parsedContent: CargoConfig;
+
+  let cargoConfig: CargoConfig | undefined;
+  if (config?.localDir) {
+    let configPath = join(config.localDir, '.cargo', 'config.toml');
+    if (!existsSync(configPath)) {
+      configPath = join(config.localDir, '.cargo', 'config');
+    }
+
+    if (existsSync(configPath)) {
+      try {
+        // TODO: fix type
```

Comment:
Comment grandfathered in from existing code, I think the type is fine personally 😅